server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/issue/ResolvedShortBranchIssuesFactory.java

```diff
@@ -0,0 +1,31 @@
+package org.sonar.server.computation.task.projectanalysis.issue;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+import org.sonar.core.issue.DefaultIssue;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.issue.IssueDto;
+import org.sonar.server.computation.task.projectanalysis.component.Component;
+import org.sonar.server.computation.task.projectanalysis.component.ShortBranchComponentsWithIssues;
+
+public class ResolvedShortBranchIssuesFactory {
+
+  private final ShortBranchComponentsWithIssues shortBranchComponentsWithIssues;
+  private final DbClient dbClient;
+
+  public ResolvedShortBranchIssuesFactory(ShortBranchComponentsWithIssues shortBranchComponentsWithIssues, DbClient dbClient) {
+    this.shortBranchComponentsWithIssues = shortBranchComponentsWithIssues;
+    this.dbClient = dbClient;
+  }
+
+  public Collection<DefaultIssue> create(Component component) {
+    try (DbSession session = dbClient.openSession(false)) {
+      return shortBranchComponentsWithIssues.getUuids(component.getKey())
+        .stream()
+        .flatMap(uuid -> dbClient.issueDao().selectResolvedOrConfirmedByComponentUuid(session, uuid).stream())
```

Comment:
Would it be a good idea to pass all uuids into the dao at once?

Reply:
There will be multiple uuids only when issues appear on multiple branches. I would have liked to assume that this will be rare in practice, but maybe that's too naive. I'll change this.